### PR TITLE
fix: avoid triggering onclick for tiles when dismissing menu [KAP-1891]

### PR DESCRIPTION
**Pull request by wejendorp**

### CODE REVIEW

File: src/blockhub/AssetInstallButton.tsx [hunk 1]
```diff
@@ -59,10 +59,22 @@ export const AssetInstallButton = (props: Props) => {
...
+            setIsInstalling(false);
+        }
+    }, []);
+
     const submenuOpen = Boolean(submenuAnchorElm);
     const assetRef = `kapeta://${props.asset.content.metadata.name}:${props.asset.version}`;
     const assetUri = parseKapetaUri(assetRef);
-    const closeSubmenu = () => {
+    const closeSubmenu = (e: SyntheticEvent) => {
+        e.stopPropagation();
```
[the remark this answers is not in the document]
Comment (wejendorp) [reply src/blockhub/AssetInstallButton.tsx:77]: There's a second place in the code where we pass `closeSubMenu` directly to the `onClick` handler. Now we also stop it there, and that seemed to do it.

File: src/blockhub/AssetInstallButton.tsx [hunk 2]
```diff
@@ -59,10 +59,22 @@ export const AssetInstallButton = (props: Props) => {
...
     const confirm = useConfirm();
     const [submenuAnchorElm, setSubmenuAnchorElm] = useState<null | HTMLElement>(null);
+
+    const [isInstalling, setIsInstalling] = useState(false);
+    const waitForInstall = useCallback(async (p: Promise<any> | undefined) => {
+        try {
+            setIsInstalling(true);
+            await p;
+        } finally {
+            setIsInstalling(false);
```
[the remark this answers is not in the document]
Comment (wejendorp) [reply src/blockhub/AssetInstallButton.tsx:69]: Hmm, yea so we need a better integration w/ the install states. There was no feedback before, when clicking the button, which felt broken imo. Let's see if we can make a proper check / indicator.

[the remark this answers is not in the document]
Comment (hofmeister) [reply src/blockhub/AssetInstallButton.tsx:69]: There is feedback right now? You sure it's not just in storybook?

[the remark this answers is not in the document]
Comment (wejendorp) [reply src/blockhub/AssetInstallButton.tsx:69]: I noticed it in my local app on friday, that nothing was happening. But maybe that was just for uninstall.
Worst case, this change is just an immediate spinner for the action, on top of the existing one that should trigger once submitted.

[the remark this answers is not in the document]
Comment (hofmeister) [reply src/blockhub/AssetInstallButton.tsx:69]: 👍 Yea my comment was more one of wording. it's not "isInstalling" - it's "isPreparing" . Future Jacob or Henrik might misinterpret what is done

[the remark this answers is not in the document]
Comment (wejendorp) [reply src/blockhub/AssetInstallButton.tsx:69]: Ah, fair. Yea better help out those guys.

**Pull request merged**